addons/stock/wizard/stock_assign_serial_numbers.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import re
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+
+
+class StockAssignSerialNumbers(models.TransientModel):
+    _name = 'stock.assign.serial.numbers'
+    _description = 'Stock Assign Serial Numbers'
+
+    product_id = fields.Many2one('product.product', 'Product', required=True)
+    move_id = fields.Many2one('stock.move', required=True)
+    initial_prod_lot_name = fields.Char('From Serial Number')
+
+    def generate_serial_numbers(self):
+        caught_initial_number = re.search("\d+", self.initial_prod_lot_name)
+        if caught_initial_number:
+            initial_number = caught_initial_number.group()
+            padding = len(initial_number)
+            splitted = re.split(initial_number, self.initial_prod_lot_name)
+            prefix = splitted[0]
+            suffix = splitted[1]
+
+            initial_number = int(initial_number)
+
+            sequence = self.env['ir.sequence'].create({
```

Comment:
why create a sequence each time ?
it is not better to link a product to sequence (add `sequence_id` in `product.product`) ? Like that you can configure SN sequence for each product, or use the default SN sequence.

Reply:
Create a sequence each time wasn't a good idea indeed...
But I think a sequence for each product isn't the purpose here (but maybe a good idea). I prefer to reuse the same default sequence each time.

Reply:
or we could do that : 
`sequence = product_id.sn_sequence_id or self.env.ref('default_product_sn_sequence')`

or in the wizard just add a sequence_id field.